lib/parallel_test_base.pm

```diff
@@ -0,0 +1,1946 @@
+    save_screenshot;
+    set_var('TEST_RUN_RESULT', '');
+    bmwqemu::save_vars();
+    bmwqemu::load_vars();
+}
+
+=head2 do_peer_initialization
+
+Initialization information about peer, save it into variables and setup passwordless
+ssh connection to it.
+=cut
+
+sub do_peer_initialization {
+    my $self = shift;
+    $self->set_test_run_progress;
+
+    record_info("Peer initialization");
+    my $_role = $self->get_parallel_role;
+    my ($_peer_info, $_peer_vars) = $self->get_peer_info(role => $_role);
+    set_var('PEER_IPADDR', $_peer_vars->{'LOCAL_IPADDR'});
+    set_var('PEER_FQDN', $_peer_vars->{'LOCAL_IPADDR'});
+    bmwqemu::save_vars();
+    bmwqemu::load_vars();
+    $self->config_ssh_pubkey_auth(addr => get_required_var('PEER_IPADDR'));
+}
+
+=head2 get_peer_info
+
+Get peer job info and variables.
+=cut
+
+sub get_peer_info {
+    my $self = shift;
+
+    my $_peer = '';
+    my $_peerid = '';
+    my $_role = $self->get_parallel_role;
+    if ($_role eq 'parent') {
+        $_peer = get_children();
+        $_peerid = (keys %$_peer)[0];
+    }
+    elsif ($_role eq 'children') {
+        $_peer = get_parents();
+        $_peerid = $_peer->[0];
+    }
+
+    my $_peerinfo = get_job_info($_peerid);
+    my $_peervars = get_job_autoinst_vars($_peerid);
+    print "Peer Job Info:", Dumper($_peerinfo);
+    print "Peer Job Vars:", Dumper($_peervars);
+    return ($_peerinfo, $_peervars);
+}
+
+=head2 config_ssh_pubkey_auth
+
+Configure SSH Public Key Authentication to host or guest. Main arguments are
+address to which ssh connects, whether overwrite (1) or not (0) existing keys,
+either host (1) or guest (0) on which operation will be done and whether die
+(1) or not (0) if any failures happen.
+=cut
+
+sub config_ssh_pubkey_auth {
+    my ($self, %args) = @_;
+    $args{addr} //= '';
+    $args{overwrite} //= 1;
+    $args{host} //= 1;
+    $args{die} //= 0;
+    croak("The address of ssh connnection must be given") if (!$args{addr});
+
+    assert_script_run("clear && ssh-keygen -b 2048 -t rsa -q -N \"\" -f ~/.ssh/id_rsa <<< y") if ($args{overwrite} == 1);
+    my $_ret = 0;
+    foreach my $_addr (split(/ /, $args{addr})) {
+        record_info("Config $_addr SSH PubKey auth");
+        next if (script_run("timeout --kill-after=3 --signal=9 15 ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root\@$_addr ls") == 0);
+        enter_cmd("clear", wait_still_screen => 3);
+        enter_cmd("ssh-copy-id -f -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ~/.ssh/id_rsa.pub root\@$_addr", wait_still_screen => 3);
+        if ($args{host} == 1) {
+            susedistribution::handle_password_prompt();
+        }
+        else {
+            check_screen("password-prompt", 60);
+            enter_cmd("novell", wait_screen_change => 50, max_interval => 1);
```

Comment:
For the guest password, better to not use explicit value here? Can be passed in by SECRET_XX.